### ISSUE TITLE
PrBoom fails to start in RetroArch v1.11.0+ on a Nintendo 2DS XL

### ISSUE THREAD

**Issue by vinceho** [libretro/RetroArch#14722]
## Description

PrBoom fails to start when prboom.cfg file exists already on a Nintendo 2DS XL device. PrBoom works fine in RetroArch v1.10.3 version. It'll give a black screen when running in 1.11.0+.  PrBoom core hasn't changed since 1.10.3, so I suspect it's a regression in the RetroArch backend.

### Expected behavior

PrBoom loads.

### Actual behavior

PrBoom fails to load and stuck on a black screen.

### Steps to reproduce the bug

1. Load Doom through PrBoom core.
2. Start a game so that the prboom.cfg file is created.
3. Quit the game and close RetroArch.
4. Restart RetroArch.
5. Load Doom again, this will now give a black screen.
6. Deleting the prboom.cfg file in the cores folder will allow the game to start properly again.

### Bisect Results
This works fine in RetroArch 1.10.3.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.11.0 and above.

### Environment information

- OS: Nintendo 2DS XL with 11.16 firmware

**Comment by LibretroAdmin** [comment]
This seems like a weird issue. So you're telling us that the Doom core works fine the first time UNTIL it creates prboom.cfg, and after that the issues start? Are you sure it might not just be some setting you are enabling that the 3DS port doesn't like ,perhaps because it's too slow to handle the setting or something?

**Comment by vinceho** [comment]
That's correct, it only fails to start with the cfg file existing. I compared the generated cfg file between v1.10.3 and v1.13.0, they're exactly the same.

**Comment by vinceho** [comment]
For more info

v1.10.3 uses PrBoom commit https://github.com/libretro/libretro-prboom/commit/b22a6b19fd976a14374db9083baea9c91b079106
v1.13.0 uses PrBoom commit https://github.com/libretro/libretro-prboom/commit/4e671fa0a4b7b892e17ac4e1803c9d627653a4c1

The main difference is some strlen optimization and some snprintf to strlcpy refactoring.
